apiserver/facades/client/spaces/spaces.go

```diff
@@ -297,7 +304,7 @@ func (api *API) ReloadSpaces(ctx stdcontext.Context) error {
 // checkSupportsSpaces checks if the environment implements NetworkingEnviron
 // and also if it supports spaces.
 func (api *API) checkSupportsSpaces(ctx stdcontext.Context) error {
-	env, err := environs.GetEnviron(ctx, api.backing, environs.New)
+	env, err := api.providerGetter(ctx)
```

Comment:
We might also want to perform a check here. If provider getter returns a not supported error it means the models provider does not support NetworkingEnviron interface. We can safely say in this case that supports space is false.

Reply:
this method only has a single error return, if the return value is not nil, then it is assumed that spaces are not supported. I don't think anything has to be changed here.

Reply:
Maybe the error message could be clearer in that case, it would look like a real error even though it might just be a cloud that doesn't support spaces

Reply:
Fair missed the single return error. Can we just add a quick documentation note to say that if we don't support spaces a `NotSupported` error will be returned?